File: backend/shared/lib/src/main/java/br/ufpr/dac/shared/dto/GerentesDto.java
package br.ufpr.dac.shared.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

public class GerentesDto {

  @Data
  @Builder
  @AllArgsConstructor
  @NoArgsConstructor
  public static class Gerente {
    private Long id;
    private String nome;
    private String email;
    private String cpf;
    private String telefone;
    private Boolean administrador;

    // construtor sem senha
    public Gerente(Long id, String nome, String email, String cpf, String telefone, Boolean administrador) {
      this.id = id;
      this.nome = nome;
      this.email = email;
      this.cpf = cpf;
      this.telefone = telefone;
      this.administrador = administrador;

    }

    private String senha;
  }

}

File: backend/apiGatewayServiceNode/src/dto/GerentesDto.ts

export interface GerentesDtoMessage {
  operation: null | string;
  data: null | GerentesDtoGerente[];
}

export interface GerentesDtoGerente {
  id: null | number;

  nome: null | string;
  email: null | string;
  cpf: null | string;
  telefone: null | string;
}
